java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/RoutingController.java

```diff
@@ -212,14 +212,16 @@ public ResponseEntity<?> routingParams() {
 					for (String profile : derivedProfilesList) {
 						rm = new RoutingMode("default".equals(profile) ? e.getKey() : profile);
 						routers.put(rm.key, rm);
-						routingService.fillRoutingModeParams(
-								Arrays.asList(hhRouting, nativeRouting, nativeTrack, sepMaps, calcMode), shortWay, e, rm);
+						List<RoutingController.RoutingParameter> passParams = "car".equals(e.getKey())
+								? Arrays.asList(hhRouting, nativeRouting, nativeTrack, sepMaps, calcMode)
+								: Arrays.asList(hhRouting, nativeRouting, nativeTrack, sepMaps);
+						routingService.fillRoutingModeParams(passParams, shortWay, e, rm);
 					}
 				} else {
 					rm = new RoutingMode(e.getKey());
 					routers.put(rm.key, rm);
 					routingService.fillRoutingModeParams(
-							Arrays.asList(hhRouting, nativeRouting, nativeTrack, sepMaps, calcMode), shortWay, e, rm);
```

Comment:
Here it looks much nicer to configure RoutingParameters, for example calculationmode could always pass to route url but not be visible (as default parameter) in url for web page

Reply:
As calcMode is nullable now, we should not require calculationmode as mandatory parameter from the web. Just done little refactor after review.